### PR TITLE
Update Raft election, lease intervals, etc.

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-6375, DOC-7125

Summary of changes:

- Update 'Distribution Layer' page with updated network timeout and mention the `COCKROACH_NETWORK_TIMEOUT` env var that controls it

- Update 'Replication Layer' page with:

  - A table listing all of the relevant intervals that can be controlled with the `COCKROACH_RAFT_ELECTION_TIMEOUT_TICKS` variable

  - More links to the table wherever the values in that table are referred to

- Finally, all of the intervals are stored in variables so their values can be referenced and updated in one place going forward

### CODE REVIEW

**Comment by rmloveland** [comment]
Hi @erikgrinaker these docs changes are based on your work in cockroachdb/cockroach#92542 and cockroachdb/cockroach#91947

Please let me know if I got some details wrong, I read through your comments in the code PRs but probably didn't understand everything properly

also open to any general feedback/comments on how we're presenting this info, or anything else that comes to mind

thanks!

File: v23.1/architecture/distribution-layer.md
Comment (erikgrinaker) [inline v23.1/architecture/distribution-layer.md:92]: This is only kind of true. gRPC network timeouts only apply from the server's point of view, not the client's (in this case the client is another CRDB node), and the client is the one that usually matters. We additionally send an RPC ping every second to check that RPC connections are alive, but these have a fairly high timeout of 6 seconds (3x network timeout) because they're subject to contention with RPC traffic (so e.g. during a large import they can get delayed by several seconds).

This additional complexity is unfortunate, but currently necessary -- ideally we would only rely on the gRPC network timeout, but gRPC doesn't let us do that from the client side.

File: v23.1/architecture/replication-layer.md
Comment (erikgrinaker) [inline v23.1/architecture/replication-layer.md:224]: This is no longer entirely true. In cockroachdb/cockroach#98616 we split out the lease interval to a separate env var `COCKROACH_RANGE_LEASE_DURATION` with a default of 6 seconds, which is independent of the Raft election timeout -- the node heartbeat interval is half of this, so 3 seconds. Similarly, cockroachdb/cockroach#98559 split out the proposal timeout to `COCKROACH_RAFT_REPROPOSAL_TIMEOUT_TICKS` defaulting to 6 ticks (3 seconds given the current 500 ms tick interval).

File: _data/constants.yml
Comment (erikgrinaker) [inline _data/constants.yml:8]: In cockroachdb/cockroach#98584, the tick interval `COCKROACH_RAFT_TICK_INTERVAL` was changed from 200 ms to 500 ms, so the Raft election timeout was reduced from 10 ticks to 4 ticks (both are equivalent to 2 seconds in wall time).

File: _data/constants.yml
Comment (erikgrinaker) [inline _data/constants.yml:2]: The election timeout is 2-4 seconds, the proposal timeout is 3 seconds (in practice 3-6 seconds).

**Review by erikgrinaker — COMMENTED** [review]
Thanks for updating this!

File: v23.1/architecture/distribution-layer.md
Comment (rmloveland) [reply v23.1/architecture/distribution-layer.md]: I added another paragraph with this info, slightly revised. PTAL since I may have misunderstood a bit

File: v23.1/architecture/replication-layer.md
Comment (rmloveland) [reply v23.1/architecture/replication-layer.md]: I've revised this table to explain that these new environment variables can be used, as well as updated the table so that the various values are shown as what they are calculated from, e.g. "n_ticks * tick_value" (where appropriate)

also updated the table to use the updated tick info (and env var) from your other comment

File: _data/constants.yml
Comment (rmloveland) [reply _data/constants.yml]: This is now updated so that the variables in the file represent the various foo_ticks variables you mentioned in other comments, and the table in replication-layer.md now shows a value something like "multiply the ticks times the interval" to describe the value

in practice it means that eg the table on `replication-layer.md` shows a value for the raft election timeout like 'N ticks * M ms' rather than a range of say "2-4s" as shown here, followed by something in the description like

> Controlled by `COCKROACH_RAFT_REPROPOSAL_TIMEOUT_TICKS`, which is then multiplied by the default tick interval to determine the value. This value is then multiplied by a random factor to avoid election ties.

**Comment by rmloveland** [comment]
@erikgrinaker this is updated based on your previous comments and should be RFAL

**Comment by erikgrinaker** [comment]
Sorry for the hold-up, I just got out of a non-stop L2 rotation and am still working my way through the backlog. Will get back to you shortly.

File: v23.1/architecture/distribution-layer.md
Comment (erikgrinaker) [reply v23.1/architecture/distribution-layer.md]: Yep, that's accurate, thanks!

File: v23.1/architecture/replication-layer.md
Comment (erikgrinaker) [inline v23.1/architecture/replication-layer.md:218]:
```suggestion
[Raft](#raft) election timeout | {{site.data.constants.cockroach_raft_election_timeout_ticks}} * {{site.data.constants.cockroach_tick_interval}} | Controlled by `COCKROACH_RAFT_ELECTION_TIMEOUT_TICKS`, which is then multiplied by the default tick interval to determine the timeout value. This value is then multiplied by a random factor of 1-2 to avoid election ties.
```

File: v23.1/architecture/replication-layer.md
Comment (erikgrinaker) [inline v23.1/architecture/replication-layer.md:219]: I'd remove the last sentence. This doesn't really have anything to do with elections, it's how often we repropose/retry a Raft command if we haven't gotten a result yet. It can technically happen within a factor of 1-2 of this, because we only check it once per reproposal interval, but I don't think we need to get that technical.

File: v23.1/architecture/replication-layer.md
Comment (erikgrinaker) [inline v23.1/architecture/replication-layer.md:221]: This is 2x the Raft election timeout (without randomization). Not sure if we need to specify that though, your call.

File: v23.1/architecture/replication-layer.md
Comment (erikgrinaker) [inline v23.1/architecture/replication-layer.md:227]: This isn't really true anymore. Only the lease acquisition timeout is now tied to the Raft election timeout, the other values are separate (arguably we should make that a separate value too, but it's too late now to do that for 23.1). I'd just nix this paragraph.

**Review by erikgrinaker — APPROVED** [review]
LGTM once comments are addressed. Thanks again for updating this!

I will say that tuning these values isn't trivial, since there are several interactions and aspects to consider. We can write something up around this if you'd like, or we can just consider this to be for advanced users who have some idea of the implications.

File: v23.1/architecture/replication-layer.md
Comment (rmloveland) [reply v23.1/architecture/replication-layer.md:221]: Will leave it out for now in service of expediency and see if someone cares.  Can always add later

File: v23.1/architecture/replication-layer.md
Comment (rmloveland) [reply v23.1/architecture/replication-layer.md:227]: Ah thanks, I think I meant to delete after latest update and missed it.  Removed entirely.

**Comment by ianjevans** [comment]
> attractive nuisance

My absolute favorite legal term.

**Review by ianjevans — APPROVED** [review]

**Pull request merged**
